CefSharp/IDownloadHandler.cs

```diff
@@ -11,17 +11,17 @@ public interface IDownloadHandler
         /// </summary>
         /// <param name="browser">The browser instance</param>
         /// <param name="downloadItem">Represents the file being downloaded.</param>
-        /// <param name="downloadPath">Path where the file will be saved if <see cref="showDialog"/> is False.</param>
-        /// <param name="showDialog">Display a dialog allowing the user to specify a custom path and filename.</param>
+        /// <param name="callback">Callback interface used to asynchronously continue a download.</param>
         /// <returns>Return True to continue the download otherwise return False to cancel the download</returns>
-        bool OnBeforeDownload(IBrowser browser, DownloadItem downloadItem, out string downloadPath, out bool showDialog);
+        void OnBeforeDownload(IBrowser browser, DownloadItem downloadItem, IBeforeDownloadCallback callback);
```

Comment:
Need to make a pass through and remove the now incorrect return XML comments on these methods.